nixos/modules/services/mail/dovecot.nix

```
@@ -828,8 +799,8 @@ in
       // optionalAttrs (cfg.group == "dovecot2") {
         dovecot2.gid = config.ids.gids.dovecot2;
       }
-      // optionalAttrs (cfg.createMailUser && cfg.mailGroup != null) {
-        ${cfg.mailGroup} = { };
+      // optionalAttrs (cfg.settings ? mail_gid && cfg.settings.mail_gid != null && cfg.createMailUser) {
```

```suggestion
      // optionalAttrs (cfg.settings.mail_gid or null != null && cfg.createMailUser) {
```

A bit less repetitive. (also above for mail_uid and mail_gid)